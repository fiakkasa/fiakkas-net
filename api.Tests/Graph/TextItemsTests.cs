using HotChocolate.Execution;

namespace api.Tests.Graph;

public class TextItemsTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task TextItems_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              textItems {
                totalCount
                nodes {
                  content
                  createdAt
                  id
                  internalId
                  key
                  title
                  updatedAt
                  version
                }
              }
            }
            """);

        var fn = result.ExpectOperationResult;

        Assert.Null(fn().Errors);
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task TextItemById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              textItemById(id: "VGV4dEl0ZW0KZzQ4ZTQ4M2U0Njk2MTRiMjU4OGE5ZDFkMGE1MTYxMTA5") {
                content
                createdAt
                id
                internalId
                key
                title
                updatedAt
                version
              }
            }
            """);

        var fn = result.ExpectOperationResult;

        Assert.Null(fn().Errors);
        result.ToJson().MatchSnapshot();
    }
}
