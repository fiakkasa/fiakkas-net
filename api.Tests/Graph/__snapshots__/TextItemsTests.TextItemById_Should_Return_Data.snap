{
  "data": {
    "textItemById": {
      "content": "Content",
      "createdAt": "2024-01-01T00:00:00.000Z",
      "id": "VGV4dEl0ZW065IPkSGFpJUuIqdHQpRYRCQ==",
      "internalId": "48e483e4-6961-4b25-88a9-d1d0a5161109",
      "key": "Key",
      "title": "Title",
      "updatedAt": null,
      "version": 1
    }
  }
}
